chain/consensus/mir/rpc/rpc.go

```diff
@@ -0,0 +1,87 @@
+package rpc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+
+	rpc "github.com/gorilla/rpc/v2/json2"
+)
+
+type JSONRPCRequestSender interface {
+	SendRequest(method string, params interface{}, reply interface{}) error
+}
+
+var _ JSONRPCRequestSender = &JSONRPCClient{}
+
+type JSONRPCClient struct {
+	ctx   context.Context
+	token string
+	url   string
+}
+
+func NewJSONRPCClient(url, token string) *JSONRPCClient {
+	return &JSONRPCClient{
+		ctx:   context.Background(),
+		token: token,
+		url:   url,
+	}
+}
+
+// NewInsecureJSONRPCClient creates a JSON RPC client with empty credentials.
+func NewInsecureJSONRPCClient(url string) *JSONRPCClient {
+	return &JSONRPCClient{
+		ctx:   context.Background(),
+		token: "",
+		url:   url,
+	}
+}
+
+func NewJSONRPCClientWithContext(ctx context.Context, url, token string) *JSONRPCClient {
+	return &JSONRPCClient{
+		ctx:   ctx,
+		token: token,
+		url:   url,
+	}
+}
+
+func (c *JSONRPCClient) SendRequest(method string, params interface{}, reply interface{}) error {
+	paramBytes, err := rpc.EncodeClientRequest(method, params)
```

Comment:
Are we sure that the method an parameters generated from this package are compatible with our implementation in the IPC agent?

Reply:
Yes, I asked @cryptoAtwill to create a string-based test in the JSON RPC server repo. Then I can use the same strings (serialized requests) to test them here.
Do you have other ideas?

Reply:
This works for now, just to double-check before we get integration tests ready.